src/clr-core/button/button.element.scss

```diff
@@ -1,35 +1,55 @@
+// Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+// This software is released under MIT license.
+// The full license information can be found in LICENSE in the root directory of this project.
+
+@import './../styles/tokens/generated/index';
+
 @import './../styles/mixins/utils';
+@import './../styles/mixins/mixins';
 
-// TODO: progress spinner CSS deprecated, will be replaced by cds-progress
+// TODO: progress spinner CSS deprecated, will be replaced by cds-circular-progress
 @import './../../clr-angular/progress/spinner/variables.spinner';
 @import './../../clr-angular/image/icons.clarity';
 @import './../../clr-angular/progress/spinner/spinner.clarity';
 
 :host {
-  --box-shadow-color: var(--clr-color-action-500, #{$clr-color-action-500});
-  --border-radius: #{$clr_baselineRem_3px};
-  --border-color: var(--clr-color-action-600, #{$clr-color-action-600});
-  --color: var(--clr-color-neutral-0, #{$clr-color-neutral-0});
-  --background: var(--clr-color-action-600, #{$clr-color-action-600});
+  --box-shadow-color: #{$cds-token-color-action-900};
+  --border-radius: #{$cds-token-global-border-radius};
+  --border-width: #{$cds-token-global-border-width};
+  --border-color: #{$cds-token-color-action-600};
+  --color: #{$cds-token-color-neutral-0};
+  --background: #{$cds-token-color-action-600};
+  --height: #{$clr_baselineRem_1_5};
   display: inline-block;
+  -webkit-appearance: none !important;
+}
+
+@mixin button-padding() {
+  padding-top: calc(var(--padding-vertical) - var(--border-width) + 0.025rem);
+  padding-left: calc(var(--padding-horizontal) - var(--border-width));
+  padding-right: calc(var(--padding-horizontal) - var(--border-width));
+  padding-bottom: calc(var(--padding-vertical) - var(--border-width) - 0.025rem);
 }
 
 .private-host {
   --font-size: #{$clr_baselineRem_0_5};
-  --line-height: 1.7rem;
+  --font-weight: 500;
   --letter-spacing: 0.12em;
-  --padding: 0 #{$clr_baselineRem_0_5};
+  --padding-vertical: #{$cds-token-layout-space-sm};
+  --padding-horizontal: #{$cds-token-layout-space-sm};
+
+  @include button-padding;
 
   background: var(--background);
   color: var(--color);
   border-color: var(--border-color);
   border-radius: var(--border-radius);
-  border-width: $clr-global-borderwidth;
+  border-width: var(--border-width);
   font-size: var(--font-size);
-  font-weight: 500;
-  line-height: var(--line-height);
+  font-weight: var(--font-weight);
+  height: var(--height);
+  line-height: 1em;
```

Comment:
We have line height here but also it is set on the slotted elements on lines 93 and 104 do we need all three or just one?

Reply:
Result of playing battleship with the LHEs. May or may not be needed anymore.

Reply:
Apparently we need two of them but not all three...